cluster-autoscaler/cloudprovider/azure/azure_instance.go

```diff
@@ -64,6 +65,13 @@ var GetVMSSTypeDynamically = func(template compute.VirtualMachineScaleSet, azCac
 	ctx := context.Background()
 	var vmssType InstanceType
 
+	// We will only use static SKU list if dynamic listing fails
+	if !azCache.HasVMSKUs() {
+		msg := "no vm sku info loaded, using only static sku list"
+		klog.Warning(msg)
+		return vmssType, errors.New(msg)
+	}
```

Comment:
It is weird to have this check _within_ `GetVMSSTypeDynamically`... Being in the very beginning, wouldn't it block the very first dynamic load?

Which got me wondering how the relevant unit test passed - and the current approach has it override `GetVMSSTypeDynamically` altogether, so it does not actually get tested. Maybe we don't have a choice but to switch to mocking skewer/resource APIs for proper test coverage ...

Reply:
The first fetch skus happens when the cache is created (`newAzureCache` in cache.go). As such, unless dynamic fails when initializing the cache, it should have skus and thus this should pass.

Reply:
anyway, I'm removing these changes from this PR for now 🤣 
but good to think about for next iteration